lib/ice_cube/builders/string_builder.rb

```diff
@@ -13,14 +13,18 @@ def piece(type, prefix = nil, suffix = nil)
     end
 
     def to_s
-      @types.each_with_object(@base || '') do |(type, segments), str|
+      string = @base || ''
+      @types.each do |type, segments|
```

Comment:
This could be done nicely with `reduce`:
```ruby
@types.reduce(@base || '') do |string, (type, segments)|
  if formatter = self.class.formatter(segments)
    current = formatter.call(segments)
  else
    next if segments.empty?
    current = self.class.sentence(segments)
  end

  format = I18n.t('ice_cube.string.format')[type] ? type : 'default'
  I18n.t("ice_cube.string.format.#{format}"), rest: string, current: current)
end
```